log-collector/src/main/scala/io/phdata/pulse/logcollector/LogCollector.scala

```diff
@@ -93,14 +93,18 @@ object LogCollector extends LazyLogging {
     }
 
     httpServerFuture.onComplete {
-      case Success(v) => ()
-      case Failure(e) => throw new RuntimeException(e)
+      case Success(v) => {
+        Await.ready(
```

Comment:
What happens when you leave the 'await' where it was?
`onComplete` is a callback so it will be called when the future finishes. So it doesn't seem right you would wait for a future to finish inside its own completion block, it will always be completed.